### PR TITLE
update misc tools

### PR DESCRIPTION
**Pull request by jacob-delgado**

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

File: docker/build-tools/Dockerfile [hunk 1]
```diff
@@ -42,13 +42,13 @@ ENV GO_JUNIT_REPORT_VERSION=df0ed838addb0fa189c4d76ad4657f6007a5811c
 ENV GOCOVMERGE_VERSION=b5bfa59ec0adc420475f97f89b58045c721d761c
 ENV GOIMPORTS_VERSION=v0.1.0
 ENV BENCHSTAT_VERSION=9c9101da8316
-ENV GH_VERSION=2.22.1
+ENV GH_VERSION=2.23.0
 ENV GOLANG_PROTOBUF_VERSION=v1.28.1
 ENV GOLANG_GRPC_PROTOBUF_VERSION=v1.2.0
 # When updating the golangci version, you may want to update the common-files config/.golangci* files as well.
 ENV GOLANGCI_LINT_VERSION=v1.51.0
```
Comment (ericvn) [inline docker/build-tools/Dockerfile:49]: I have this updated to 1.51.1 as well.

**Pull request merged**